skyportal/handlers/api/gwdetector.py

```diff
@@ -0,0 +1,234 @@
+from marshmallow.exceptions import ValidationError
+from baselayer.app.access import permissions, auth_or_token
+
+from ..base import BaseHandler
+from ...models import GWDetector
+
+
+class GWDetectorHandler(BaseHandler):
+    @permissions(['Manage allocations'])
+    def post(self):
+        """
+        ---
+        description: Create gwdetector
+        tags:
+          - gwdetectors
+        requestBody:
+          content:
+            application/json:
+              schema: GWDetectorNoID
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            id:
+                              type: integer
+                              description: New gwdetector ID
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        data = self.get_json()
+
+        with self.Session() as session:
+
+            schema = GWDetector.__schema__()
+            if 'lat' not in data or 'lon' not in data:
+                return self.error('Missing latitude or longitude')
+            elif not isinstance(data['lat'], (int, float)) or not isinstance(
+                data['lon'], (int, float)
+            ):
+                return self.error('Latitude and longitude must all be numbers')
```

Comment:
Isn't this taken care of by the schema validation?

Reply:
Well it catches it pre-schema. You want it gone?

Reply:
If it is indeed redundant I suggest we remove it. Will save you some time in the PUT/PATCH handler.

Reply:
done.